ocf_datapipes/transform/xarray/add_contiguous_time_periods.py

```diff
@@ -0,0 +1,103 @@
+import datetime
```

Comment:
I might need some motivation why we want to do this

Reply:
This is how we get all the time periods in nowcasting_dataset and PP training to sample from, so we don't sample where there is missing timesteps, more for training than productions.